parsely/src/main/java/com/parsely/parselyandroid/ParselyTracker.java

```diff
@@ -102,6 +105,14 @@ protected ParselyTracker(String siteId, int flushInterval, Context c) {
         if (getStoredQueue().size() > 0) {
             startFlushTimer();
         }
+
+        ProcessLifecycleOwner.get().getLifecycle().addObserver(
+                (LifecycleEventObserver) (lifecycleOwner, event) -> {
+                    if (event == Lifecycle.Event.ON_STOP) {
+                        new FlushQueue().execute();
```

Comment:
Suggestion (💡): Since there are at least two instances of using this `new FlushQueue().execute();` call, maybe a new private `flushEventQueue()` method should be created and used instead.